packages/client/hooks/useTransition.ts

```diff
@@ -22,6 +22,16 @@ const useTransition = <T extends {key: Key}>(children: T[]) => {
   const previousTransitionChildrenRef = useRef<TransitionChild<T>[]>([])
   const forceUpdate = useForceUpdate()
 
+  useEffect(() => {
```

Comment:
can you give me a test scenario so i can put this to the test? not sure what it does

Reply:
For sure! 

1. Comment out the useEffect hook I've added here in `useTransition` 
2. Remove the `isInit` check [here](https://github.com/ParabolInc/parabol/blob/e04f3fb9ed3fff3b4fb78d44ee9b818397c156c8/packages/client/hooks/useOverflowAvatars.ts#L36) 
3. Console log the transitioning avatars [here](https://github.com/ParabolInc/parabol/blob/e04f3fb9ed3fff3b4fb78d44ee9b818397c156c8/packages/client/hooks/useOverflowAvatars.ts#L49)
4. Open two windows. In window A, join an active meeting and see that the connected user is added to the `AvatarList` on the meeting card in window B
5. In window B, refresh the page and see that the transitioning avatars console log includes both the connected user and the overflow avatar
6. Replace `activeTChildren` with `transitioningChildren` in AvatarList [here](https://github.com/ParabolInc/parabol/blob/f21a82d9be54f1cbb2c6412e77b2ae45651306d9/packages/client/components/AvatarList.tsx#L57) and leave the meeting in window A. Even though there are no longer any connected users, you'll see that the meeting card height doesn't shrink as it recognises the overflow avatar with status `EXITING` as a connected user

By adding this clean-up function, I'm removing items that have a status of `EXITING` when the user refreshes the page.  The `isInit` check in `useOverflowAvatars` ensures that the `overflowCount` calculation is accurate even on the first render.

Reply:
if i understand correctly, the root cause is that the overflow avatar is showing up invisibly when it shouldn't.

if useOverflowAvatars uses the following line, can you remove this hook?
`  const overflowCount = maxAvatars > 0 && totalItems > maxAvatars ? totalItems - maxAvatars + 1 : 0`

Reply:
Yep, that'll work for this use case but if a transitioningChild has a status of `EXITING` and is no longer needed, wouldn't we want to clean it up (based on [this suggestion](https://github.com/ParabolInc/parabol/pull/4975#discussion_r643562397))?

Reply:
While the clean-up function might be useful in the future, it doesn't solve any existing issue right now so I've removed it